### PR TITLE
Add `CITATION.cff` file

### PR DESCRIPTION
**Pull request by drupol**
Hello,

I frequently need to cite Typst in the documents I write, and noticed that GitHub supports a specific citation file format (`CITATION.cff`). Having this file would enhance the visibility and proper citation of Typst directly from the GitHub repository page.

GitHub's support for the `CITATION.cff` files enables easy access to citation information, which can be automatically parsed and displayed on the project page. More details can be found in the [GitHub documentation on citation files](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files).

Would you be interested in adding such a file to the project?

When Github detect such a file in a repo, a new link is displayed on the right side:

image

Clicking on the link open a box:

image

Screenshots from https://github.com/citation-file-format/citation-file-format

Related to https://github.com/typst/typst/discussions/4108

### CODE REVIEW

**Comment by drupol** [comment]
See it lively at https://github.com/drupol/typst/tree/add-citation-cff

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+abstract: >-
+  Typst was initiated in 2019 at the Technical University of
+  Berlin by Martin Haug and Laurenz Mädje. Developed in
+  Rust, this programmable markup language for typesetting
+  became the subject of their master's theses, which they
+  wrote in 2022. After several years of closed-source
+  development, Typst was open-sourced and released to the
+  public in 2023.
+keywords:
+  - type-setting
```
Comment (xkevio) [inline CITATION.cff:33]:
```suggestion
  - typesetting
```

**Comment by xkevio** [comment]
Would it also maybe make sense to add the `date-released` field? As like the year the repository went public? It's not ideal, especially once Typst is out of beta but citing something without some form of year can look off in most citation styles.

**Comment by drupol** [comment]
> Would it also maybe make sense to add the `date-released` field? As like the year the repository went public? It's not ideal, especially once Typst is out of beta but citing something without some form of year can look off in most citation styles.

Yes ! The more field we have, the better ! feel free to suggest them !!!

**Comment by xkevio** [comment]
> > Would it also maybe make sense to add the `date-released` field? As like the year the repository went public? It's not ideal, especially once Typst is out of beta but citing something without some form of year can look off in most citation styles.
> 
> Yes ! The more field we have, the better ! feel free to suggest them !!!

On that note, the `version` field could be nice but would have to be updated alongside releases.

**Comment by drupol** [comment]
The `date-released` field is usually used when we add the versions inside that file. TBH, I don't believe we should add the versions, therefore, I don't think this field is really needed.

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+repository-code: 'https://github.com/typst/typst'
+url: 'https://typst.app/'
+abstract: >-
+  Typst was initiated in 2019 at the Technical University of
+  Berlin by Martin Haug and Laurenz Mädje. Developed in
+  Rust, this programmable markup language for typesetting
+  became the subject of their master's theses, which they
+  wrote in 2022. After several years of closed-source
+  development, Typst was open-sourced and released to the
+  public in 2023.
```
Comment (laurmaedje) [inline CITATION.cff:31]: The text here works fine as a paragraph about Typst in a background section, but I'm not sure it's great as an abstract. I would prefer to just drop the abstract. I don't think it's ever needed for a citation.

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  If you use this software, please cite it using the
+  metadata from this file.
+type: software
+authors:
+  - given-names: Laurenz
+    family-names: Mädje
+  - given-names: Martin
+    family-names: Haug
+  - name: The Typst Project Developers
+    email: hello@typst.app
```
Comment (laurmaedje) [inline CITATION.cff:16]:
```suggestion
  - given-names: Laurenz
    family-names: Mädje
    email: laurenz.maedje@typst.app
  - given-names: Martin
    family-names: Haug
    email: martin.haug@typst.app
  - name: The Typst Project Developers
```

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+identifiers:
+  - type: url
+    value: >-
+      https://laurmaedje.github.io/programmable-markup-language-for-typesetting.pdf
+    description: A Programmable Markup Language for Typesetting
```
Comment (laurmaedje) [inline CITATION.cff:21]:
```suggestion
    description: "Typst: A Programmable Markup Language for Typesetting"
```

**Comment by drupol** [comment]
I applied your suggestions, and also added a new CI file to check the validity of the `CITATION.cff` file.

**Review by laurmaedje — COMMENTED** [review]
I think CI for a citation file is overkill, so I'd rather not have that. 🙂

**Comment by drupol** [comment]
Removed

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**